### ISSUE TITLE
Knot plot() returns questionable non-planar knot diagram (moreover, with overlapping segments)

### ISSUE THREAD

**Issue by PratherConid** [sagemath/sage#37587]
### Steps To Reproduce

```
kn = Knots().from_dowker_code([30, 18, 20, 24, 22, 26, 28, 32, 2, 4, 6, 8, 12, 10, 16, 14])
kn.plot()
```

### Expected Behavior

A knot diagram, which should be a planar graph with gaps at each vertex indicating over-crossing/under-crossing.

### Actual Behavior

The `plot` function returns a knot diagram corresponding to a non-planar graph, and with overlapping segments.
![image](https://github.com/sagemath/sage/assets/58621828/88859207-a55c-4a74-b52c-b22d5eda4697)

### Environment

```markdown
- **OS**: Ubuntu 20.04
- **Sage Version**: 10.2
```

### Checklist

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [X] I have read the documentation and troubleshoot guide

**Comment by tscrim** [comment]
@soehms

**Comment by soehms** [comment]
The good news: There is a workaround which produces a correct result in the case of the knot in question:

```
kn = Knots().from_dowker_code([30, 18, 20, 24, 22, 26, 28, 32, 2, 4, 6, 8, 12, 10, 16, 14])
sage: kn.plot(solver='PPL')
```

![kn_ppl](https://github.com/sagemath/sage/assets/47305845/6ce24490-2e1c-492f-98b1-d50a4ba8de02)

The bad news: I still need time to fully understand the implementation of this method. It uses mixed integer linear programming twice. The problem occurs in the first step, which is determining the number of bends required for the segments. The constraints on the MLP at this step are probably not strong enough. I suspect that this is due to taking the exterior region out of consideration:

```
        regions = sorted(self.regions(), key=len)
        regions = regions[:-1]
```


@amitjamadagni, @miguelmarco do you remember the ideas behind this?

**Comment by DaveWitteMorris** [comment]
I think this issue should be closed, because it was fixed by #37851.

**Comment by tscrim** [comment]
Indeed.
